### ISSUE TITLE
Carousel: tab navigation breaks when slides have focusable elements

### ISSUE THREAD

**Issue by mayank99** [iTwin/iTwinUI#912]
#### Describe the bug (actual behavior)

When Carousel slides have focusable elements inside, then keyboard navigation breaks. Pressing <kbd>Tab</kbd> repeatedly cycles between two slides and makes it impossible to move forward in tab navigation.

#### Expected behavior

Keyboard navigation should behave normally. See bullet points below for expected behavior _(in italics)_ in context of repro.

#### Reproduction
Link to a minimal repro: https://codesandbox.io/s/carousel-focus-trap-bug-xkfwsf?

##### Steps to reproduce
1. Click anywhere in viewport, then press <kbd>Tab</kbd> until the Carousel as a whole is focused.
2. Press <kbd>Tab</kbd> again to focus button that reads `1`.
3. Press <kbd>Tab</kbd> again to see slide is advanced and the Carousel is focused again. _(This should have instead either focused the next button `2` or should have focused the button that is outside the carousel)._
4. Press <kbd>Tab</kbd> again to see slide is reset back to first one. _(This should have either focused button `3` or the next button outside the carousel)_

#### Additional information

This issue is likely because of this line:
https://github.com/iTwin/iTwinUI-react/blob/628cc26afc044e1b3f1c09b2489b31fe15b34de1/packages/iTwinUI-react/src/core/Carousel/Carousel.tsx#L78
